node/src/Pos/Launcher/Resource.hs

```diff
@@ -244,7 +245,7 @@ allocateNodeContext
     -> InitMode ssc (NodeContext ssc)
 allocateNodeContext np@NodeParams {..} sscnp putSlotting networkConfig = do
     ncLoggerConfig <- getRealLoggerConfig $ bpLoggingParams npBaseParams
-    ncBlkSemaphore <- BlkSemaphore <$> newEmptyMVar
+    ncStateLock <- StateLock <$> newEmptyMVar <*> newPrioLock
```

Comment:
Minor: I think it would be good to have `newStateLock` defined in `Pos.StateLock` and use it here. Maybe later we will need to create it in other places (in tests or anywhere).

Reply:
I agree. The StateLock should be a full abstraction.

Reply:
Good catch, thanks.